api/api.go

```diff
@@ -0,0 +1,190 @@
+package api
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/contiv/errored"
+	"github.com/contiv/volplugin/config"
+	"github.com/contiv/volplugin/errors"
+	"github.com/contiv/volplugin/lock"
+	"github.com/contiv/volplugin/storage/control"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// VolumeRequest is taken from
+// https://github.com/calavera/docker-volume-api/blob/master/api.go#L23
+type VolumeRequest struct {
```

Comment:
I think docker has these structures defined as well (https://github.com/docker/docker/blob/master/vendor/src/github.com/docker/engine-api/types/types.go#L437), may be we should use those or vendor them in to avoid any incompatibilities between structs.